### PR TITLE
Gvsoc pulp target rebased

### PR DESCRIPTION
**Pull request by PhilippvK**

### CODE REVIEW

**Comment by PhilippvK** [comment]
@zhouxinyu0723 I added the missing imports to `pulp.py`, feel free to continue from here and I will close #90.

File: mlonmcu/target/riscv/gvsoc_pulp.py [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+        metrics.add("Cycles", cycles)
+        metrics.add("Instructions", instructions)
+
+        return metrics, out, []
+
+    def get_target_system(self):
+        if self.model == "pulp":
+            return "gvsoc_pulp"
+        if self.model == "pulpissimo":
+            return "gvsoc_pulpissimo"
```
Comment (PhilippvK) [inline mlonmcu/target/riscv/gvsoc_pulp.py:333]: I would prefer `f"gvsoc_{self.model}"`

File: mlonmcu/target/riscv/gvsoc_pulp.py [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+
+    def get_platform_defs(self, platform):
+        assert platform == "mlif"
+        ret = super().get_platform_defs(platform)
+        # ret["RISCV_ARCH"] = "rv32imcxpulpv3"
+        ret["RISCV_ABI"] = self.abi
+
+        # The commented code below attempted to use the GCC low-level runtime library (libgcc) come with the pulp-gcc
```
Comment (PhilippvK) [inline mlonmcu/target/riscv/gvsoc_pulp.py:341]: Remove comments

File: mlonmcu/target/riscv/gvsoc_pulp.py [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+            else:
+                extra_args = self.extra_args
+            gvsoc_simulating_arg.extend(extra_args)
+
+        env = os.environ.copy()
+        env.update({"PULP_RISCV_GCC_TOOLCHAIN": str(self.pulp_gcc_prefix)})
+
+        # run simulation
+        env = os.environ.copy()
+        env.update({"PULP_RISCV_GCC_TOOLCHAIN": str(self.pulp_gcc_prefix)})
```
[the remark this answers is not in the document]
Comment (zhouxinyu0723) [reply mlonmcu/target/riscv/gvsoc_pulp.py:290]: I have removed the duplicated one.

**Comment by PhilippvK** [comment]
Rebased down to 25 commits

**Comment by PhilippvK** [comment]
@zhouxinyu0723 Could you please clean up `resources/pycharm/.run` a little bit to only have one file for `init`, `setup` and `run`? A very short `README.md` explaining how to use those with Pycharm would be nice, too.

**Comment by zhouxinyu0723** [comment]
Sorry I forgot that, sure. Should i create another branch, or do it on the gcsocpulp brnach?

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by PhilippvK** [comment]
Just create a new branch based in current main

Xinyu Zhou ***@***.***> schrieb am Mi. 21. Dez. 2022 um 11:31:

**Pull request merged**